README.md

```diff
@@ -1,16 +1,241 @@
+
+Ultimately, in order to model a real grid-based vending machine (C20 is a candy bar
+gum, B50 is a drink), we think the right NDS is:
+
+```text
+Array of Array
+...of Array
+...of Hashes
+...with keys `:name` and `:pieces` where
+......:name points to a `String`
+......:pieces points to an Integer count of pieces
+```
+
+We're giving you the answer so that you can think about how this might work for
+yourself. Draw it out on paper and see if you can predict how our explanation
+will work. Teach your friend, your cat, or your imaginary friend Hogarth how ***you***
+think things work **and then** read the explanation below. Don't lose this chance
+to train your brain!
+
+### Explaining the Vending Machine
+
+Many vending machines use a grid to identify which snack one wants. The keyword
+"grid" should immediately suggest an AoA.  Using a coordinate like `[3][2]` we
+might pick the third element on the 4th row (remember, indexes start at 0 in a
+Ruby vending machine).
+
+### Explaining the Coordinate to Array
+
+Here's a picture of what we just described in code:
+
+```ruby
+vending_machine = [
+   [
+     ["Vanilla Cookies", "Pistachio Cookies", "Chocolate Cookies", "Chocolate Chip Cookies"],
+     ["Tooth-Melters", "Tooth-Destroyers", "Enamel Eaters", "Dentist's Nighmare"],
+     ["Gummy Apple", "Gummy Apple", "Gummy Moldy Apple"]
+   ],
+   [
+     ["Grape Drink", "Orange Drink", "Pineapple Drink"],
+     ["Mints", "Curiously Toxic Mints", "US Mints"]
+    ]
+  ]
+#=> [[["Vanilla Cookies", "Pistachio Cookies", "Chocolate Cookies", "Chocolate Chip Cookies"], ["Tooth-Melters", "Tooth-Destroyers", "Enamel Eaters", "Dentist's Nighmare"], ["Gummy Apple", "Gummy Apple", "Gummy Moldy Apple"]], [["Grape Drink", "Orange Drink", "Pineapple Drink"], ["Mints", "Curiously Toxic Mints", "US Mints"]]]
+```
+
+### Explaining the Coordinate as Pointing to an Array
+
+This first step was good, but it misses an important detail about 
+_real_ vending machines: most grid-based vending machines actually have multiple instances of the
+snack at that coordinate. The structure above has _only one_
+`String` at each coordinate. Also, each snack at a coordinate has some
+properties. This model isn't quite accurate enough.
+
+Most often, each coordinate points to a corkscrew-shaped spinner that rotates
+enough to "pop" the first snack container off the spinner. This suggests that
+each coordinate should point to an `Array` instead of a simple `String`. Thus
+we're up to an "AoAoA" (`Array` of `Array`s of `Array`s).
+
+### Explaining Each Snack Items as Hashes
+
+So in the `Array` that represents the "spinner" device, how should we represent
+each snack item?
+
+A simple `String` would tell us what the snack's name is ***and that might
+be good enough for your purposes***! But when _we_ close our eyes and imagine
+a snack, we imagine a _name_ for the snack and a count of _pieces_, together.
+
+We'll leave out the price for the moment. Snacks in our vending machine
```

Comment:
eh I still think price is a clearer attribute and more relevant to the vending machine example.